web/src/inbox_ui.js

```diff
@@ -0,0 +1,875 @@
+    return "stream_" + stream_id;
+}
+
+function get_stream_container(stream_key) {
+    return $(`#${CSS.escape(stream_key)}`);
+}
+
+function get_topics_container(stream_id) {
+    const $topics_container = get_stream_header_row(stream_id).next();
+    return $topics_container;
+}
+
+function get_stream_header_row(stream_id) {
+    const $stream_header_row = $(`#${CSS.escape(STREAM_HEADER_PREFIX + stream_id)}`);
+    return $stream_header_row;
+}
+
+function load_filter() {
+    include_muted = ls.get(ls_key);
+    update_filters();
+}
+
+function update_filters() {
+    const $mute_checkbox = $("#inbox_filters #inbox_filter_mute_toggle");
+    const $mute_filter = $("#inbox_filters .btn-inbox-filter");
+    if (include_muted) {
+        $mute_checkbox.removeClass("fa-square-o");
+        $mute_checkbox.addClass("fa-check-square-o");
+        $mute_filter.addClass("btn-inbox-selected");
+    } else {
+        $mute_checkbox.removeClass("fa-check-square-o");
+        $mute_checkbox.addClass("fa-square-o");
+        $mute_filter.removeClass("btn-inbox-selected");
+    }
+}
+
+export function toggle_filter() {
+    const $mute_filter = $("#inbox_filters .btn-inbox-filter");
+    if ($mute_filter.hasClass("btn-inbox-selected")) {
+        include_muted = false;
+    } else {
+        include_muted = true;
+    }
+
+    update_filters();
+    save_filter();
+    update_data_and_ui();
+}
+
+function format_pm(user_ids_string) {
+    const recipient_ids = people.user_ids_string_to_ids_array(user_ids_string);
+    if (!recipient_ids.length) {
+        // Self DM
+        recipient_ids.push(people.my_current_user_id());
+    }
+
+    const unread_count = unread.num_unread_for_user_ids_string(user_ids_string);
+    const reply_to = people.user_ids_string_to_emails_string(user_ids_string);
+    const recipients_info = [];
+
+    for (const user_id of recipient_ids) {
+        const recipient_user_obj = people.get_by_user_id(user_id);
+        const user_circle_class = buddy_data.get_user_circle_class(user_id);
+        recipient_user_obj.user_circle_class = user_circle_class;
+
+        if (recipient_user_obj.is_bot) {
+            // Bots do not have status emoji, and are modeled as
+            // always present. We may want to use this space for a
+            // bot icon in the future.
+            recipient_user_obj.user_circle_class = "user_circle_green";
+        }
+        recipients_info.push(recipient_user_obj);
+    }
+    const context = {
+        conversation_key: user_ids_string,
+        is_private: true,
+        recipients_info,
+        pm_url: hash_util.pm_with_url(reply_to),
+        user_ids_string,
+        unread_count,
+        is_hidden: filter_should_hide_row({pm_key: user_ids_string}),
+    };
+
+    return context;
+}
+
+function validate_pm_data(new_pm_data, old_pm_data) {
+    // This row is not rendered yet.
+    if (old_pm_data === undefined) {
+        $("#pm_container").append(render_inbox_row(new_pm_data));
```

Comment:
I think the comment here can be pushed down into the if statement, and a new comment could go at the top of this function explaining its semantics. Though maybe we don't need a comment and just a new name; `rerender_dm_inbox_row_if_needed(new_data, old_data)`.

Reply:
Done.